### ISSUE TITLE
Exiting Docker shell failed

### ISSUE THREAD

**Issue by binkley** [charleskorn/batect#46]
Using batect 0.24.0.  Upgraded from 0.23.0 by copying a newer shell script version over the older one.  This task:
```
tasks:
    shell:
        run:
            container: build-env
            command: bash --login
```
Produced this error when exiting:
```
build-env: running bash --login

boxley@build-env:/src$ exit
logout

Error: An unexpected exception occurred during execution.
java.lang.RuntimeException: Invoking 'stty 6b02:3:4b00:200005cb:4:ff:ff:7f:17:15:12:ff:3:1c:1a:19:11:13:16:f:1:0:14:ff' failed with exit code 1: stty: 'standard input': unable to perform all requested operations

Clean up: done

The task shell failed. See above for details.
```

**Comment by charleskorn** [comment]
A couple of questions:

1. Do you consistently see this issue now with v0.24.0, but never see it with v0.23.0? How about with 0.23.3? (0.23.3 had a change related to this)

2. Can you please try running `stty -g` and then `stty <output from stty -g>` outside of batect (ie. in a standard Bash shell) - do you see the same error? 
  
   (So if `stty -g` gives you `abc123:456`, please run `stty abc123:456` as the second command.)

3. Does this affect other commands? For example, if your container runs `echo Hello World` instead of `bash --login`, do you see the same issue?

4. What OS are you running?

**Comment by binkley** [comment]
I tested different versions by globally replacing the version string in my latest copy of the `batect` script.  I need to check if that script itself changed.

Output from 0.24.0:
```
build-env: running bash --login

boxley@build-env:/src$ eexxiitt^M
logout

Error: An unexpected exception occurred during execution.
java.lang.RuntimeException: Invoking 'stty 6b02:3:4b00:200005cb:4:ff:ff:7f:17:15:12:ff:3:1c:1a:19:11:13:16:f:1:0:14:ff' failed with exit code 1: stty: 'standard input': unable to perform all requested operations

Clean up: done
```
(Note: something is whacky.  My typing is echoed twice for "exit".)

Output from 0.23.3:
```
build-env: running bash --login

boxley@build-env:/src$ exit
logout

Error: An unexpected exception occurred during execution.
java.lang.RuntimeException: Invoking 'stty 6b02:3:4b00:200005cb:4:ff:ff:7f:17:15:12:ff:3:1c:1a:19:11:13:16:f:1:0:14:ff' failed with exit code 1: stty: 'standard input': unable to perform all requested operations

Clean up: done

The task shell failed. See above for details.
```
(Note: no doubling of my characters in "exit".)

**Comment by charleskorn** [comment]
Are you sure the output you've given there is around the right way? The reason I ask is the issue with double-echoing of text was fixed in v0.23.3.

But regardless, it looks like the issue appears in both versions, so I'll investigate further.

**Comment by charleskorn** [comment]
Duplicate of #51.
